From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Simental Magana, Marcos" <marcos.simental.magana@intel.com>
Date: Wed, 24 Jan 2018 11:17:28 -0600
Subject: [PATCH] clearlinux stateless configuration

Changes for the 1.4.21 release: fall back to /usr/share/defaults/mock
for bash completions and for the logging.ini file.

Signed-off-by: Simental Magana, Marcos <marcos.simental.magana@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 mock/docs/mock.1                |  9 ++++---
 mock/etc/bash_completion.d/mock |  4 ++--
 mock/etc/mock/clear.cfg         | 42 +++++++++++++++++++++++++++++++++
 mock/py/mock.py                 |  9 ++++---
 4 files changed, 56 insertions(+), 8 deletions(-)
 create mode 100644 mock/etc/mock/clear.cfg

diff --git a/mock/docs/mock.1 b/mock/docs/mock.1
index b5b07cd..91b4b9f 100644
--- a/mock/docs/mock.1
+++ b/mock/docs/mock.1
@@ -57,8 +57,9 @@ contents specified by a configuration file, then build any input SRPM(s) in
 that chroot.
 .LP
 The content of a chroot is specified by the configuration specified with the
-\fB\-r\fR option. The default configuration file is /etc/mock/default.cfg,
-which is usually a symlink to one of the installed configurations.
+\fB\-r\fR option. The default configuration file is
+/usr/share/defaults/mock/default.cfg, which is usually a symlink to one
+of the installed configurations.
 .LP
 There is a site\-wide configuration file, /etc/mock/site\-defaults.cfg, which can
 be used to specify site\-wide options. The shipped version of this file has no
@@ -435,7 +436,9 @@ don't create bootstrap chroot from container image
 
 .SH "FILES"
 .LP
-\fI/etc/mock/\fP \- default configuration directory
+\fI/usr/share/defaults/mock/\fP \- default configuration directory (used when /etc/mock/ is missing)
+.LP
+\fI/etc/mock/\fP \- system configuration directory
 .LP
 \fI/var/lib/mock\fP \- directory where chroots and results are created. You should not put there your input files.
 .SH "EXAMPLES"
diff --git a/mock/etc/bash_completion.d/mock b/mock/etc/bash_completion.d/mock
index 87cdcd4..9f5c3d2 100644
--- a/mock/etc/bash_completion.d/mock
+++ b/mock/etc/bash_completion.d/mock
@@ -2,7 +2,7 @@
 
 _mock_root()
 {
-    test $# -eq 0 && set -- /etc/mock "$HOME/.config/mock"
+    test $# -eq 0 && set -- /etc/mock "$HOME/.config/mock" /usr/share/defaults/mock
 
     local suggest=$(
       shopt -s nullglob
@@ -28,7 +28,7 @@ _mock()
     local cur prev words cword split
     _init_completion -s || return
 
-    local cfgdirs=( /etc/mock "$HOME/.config/mock" )
+    local cfgdirs=( /etc/mock "$HOME/.config/mock" /usr/share/defaults/mock )
     local count=0
     for word in "${words[@]}" ; do
         [[ $count -eq $cword ]] && break
diff --git a/mock/etc/mock/clear.cfg b/mock/etc/mock/clear.cfg
new file mode 100644
index 0000000..49e866b
--- /dev/null
+++ b/mock/etc/mock/clear.cfg
@@ -0,0 +1,42 @@
+config_opts['root'] = 'clear'
+config_opts['target_arch'] = 'x86_64'
+config_opts['legal_host_arches'] = ('x86_64',)
+config_opts['chroot_setup_cmd'] = 'groupinstall build srpm-build'
+config_opts['dist'] = 'clear'  # only useful for --resultdir variable subst
+config_opts['extra_chroot_dirs'] = [ '/run/lock', ]
+config_opts['useradd'] = '/usr/sbin/useradd -m -u %(uid)s -g %(gid)s -d %(home)s  %(user)s'
+config_opts['plugin_conf']['ccache_enable'] = False
+config_opts['plugin_conf']['root_cache_opts']['compress_program'] = 'zstd -5'
+config_opts['plugin_conf']['root_cache_opts']['extension'] = '.zst'
+config_opts['releasever'] = 'clear'
+config_opts['package_manager'] = 'dnf'
+config_opts['chrootgroup'] = 'mockbuild'
+config_opts['use_bootstrap_container'] = False
+
+config_opts['dnf.conf'] = """
+[main]
+cachedir=/var/cache/dnf
+keepcache=1
+debuglevel=1
+reposdir=/dev/null
+logfile=/var/log/dnf.log
+retries=20
+obsoletes=1
+gpgcheck=0
+assumeyes=1
+syslog_ident=mock
+syslog_device=
+# repos
+[clearlinux]
+name=clearlinux
+baseurl=https://cdn.download.clearlinux.org/current/x86_64/os/
+gpgcheck=0
+cost=2000
+enabled=1
+[debuginfo]
+name=debuginfo
+baseurl=https://cdn.download.clearlinux.org/current/x86_64/debug/
+gpgcheck=0
+cost=2000
+enabled=1
+"""
diff --git a/mock/py/mock.py b/mock/py/mock.py
index 251a080..5b17563 100755
--- a/mock/py/mock.py
+++ b/mock/py/mock.py
@@ -72,6 +72,7 @@ SYSCONFDIR = os.path.join(os.path.dirname(os.path.realpath(sys.argv[0])), "..",
 PYTHONDIR = os.path.dirname(os.path.realpath(sys.argv[0]))
 PKGPYTHONDIR = os.path.join(os.path.dirname(os.path.realpath(sys.argv[0])), "mockbuild")
 MOCKCONFDIR = os.path.join(SYSCONFDIR, "mock")
+DEFAULTMOCKCONFDIR = "/usr/share/defaults/mock"
 # end build system subs
 
 # import all mockbuild.* modules after this.
@@ -435,10 +436,10 @@ def setup_logging(config_path, config_opts, options):
 
     try:
         if not os.path.exists(log_ini):
-            if os.path.normpath('/etc/mock') != os.path.normpath(config_path):
+            if os.path.normpath(DEFAULTMOCKCONFDIR) != os.path.normpath(config_path):
                 log.warning("Could not find required logging config file: %s. Using default...",
                             log_ini)
-                log_ini = os.path.join("/etc/mock", config_opts["log_config_file"])
+                log_ini = os.path.join(DEFAULTMOCKCONFDIR, config_opts["log_config_file"])
                 if not os.path.exists(log_ini):
                     raise IOError("Could not find log config file %s" % log_ini)
             else:
@@ -645,7 +646,9 @@ def main():
         options.verbose = 0
 
     # config path -- can be overridden on cmdline
-    config_path = MOCKCONFDIR
+    config_path = DEFAULTMOCKCONFDIR
+    if os.path.exists(MOCKCONFDIR):
+        config_path = MOCKCONFDIR
     if options.configdir:
         config_path = options.configdir
 
